### ISSUE TITLE
Twisted Names example testdns.py presents results very poorly

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5594]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5594|
|Type|enhancement|
|Created|2012-04-02 14:23:30Z|

Some problems:

  1. Arbitrary 4 second timeout.  If results don't arrive by then, the example silently exits (thanks, slow network, for pointing this out to me)
  1. When results do come back, they only the `RRHeader` instances are printed.  This excludes actual address information, making it difficult or impossible to actually see the core part of the functionality these APIs are supposed to provide.

It would be great to:

  1. run until the queries all succeed or fail
  1. display the `RRHeader` data _and_ the type-specific payload

Attachments:

* [testdns-output-5594.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ed5/ed573491383d5d7052276dd09beebea1637ac2a3/22d625e9158cdf2ccb42edf88273af33096cc4d0.patch) (2722 bytes) - added by rwall on 2012-09-02 22:18:20Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5594 5594
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__ 
branch_author__rwall rwall
status__closed closed
resolution__fixed fixed
component__names names
keywords__examples_documentation examples documentation
time__1333376610000000 1333376610000000
changetime__1347140639000000 1347140639000000
version__None None
owner__thijs thijs
cc__rwall cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_1"></a>|@wallrj commented|
|-|-|

Added a patched version of testdns.py with better output and which exits when all results are in.

Not sure whether a topfile is necessary for a documentation change?

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

Much better! `python doc/names/examples/testdns.py nsa.gov`:

```
Addresses:  [<RR name=nsa.gov type=A class=IN ttl=86362s auth=False>, <RR name=nsa.gov type=A class=IN ttl=86362s auth=False>, <RR name=nsa.gov type=A class=IN ttl=86362s auth=False>, <RR name=nsa.gov type=A class=IN ttl=86362s auth=False>], [], []
Nameservers:  [<RR name=nsa.gov type=NS class=IN ttl=28762s auth=False>, <RR name=nsa.gov type=NS class=IN ttl=28762s auth=False>], [], []
Mail Exchangers:  [<RR name=nsa.gov type=MX class=IN ttl=3600s auth=False>, <RR name=nsa.gov type=MX class=IN ttl=3600s auth=False>], [], []
```

vs.

```
# Addresses
nsa.gov IN <A address=12.120.172.8 ttl=86400>
nsa.gov IN <A address=12.120.166.8 ttl=86400>
nsa.gov IN <A address=12.120.186.8 ttl=86400>
nsa.gov IN <A address=12.120.184.8 ttl=86400>

# Mail Exchangers
nsa.gov IN <MX preference=20 name=emvm-gh1-uea08.nsa.gov ttl=3600>
nsa.gov IN <MX preference=20 name=emvm-gh1-uea09.nsa.gov ttl=3600>

# Nameservers
nsa.gov IN <NS name=dsdn-gh1-uea05.nsa.gov ttl=28800>
nsa.gov IN <NS name=dsdn-gh1-uea06.nsa.gov ttl=28800>
```

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [35558]) Apply testdns-output-5594.patch.

Author: rwall
Reviewer: thijs
Fixes: [#5594](https://github.com/twisted/twisted/issues/5594)

Pretty print the output of the testdns example.
